flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/snapshot/RocksIncrementalSnapshotStrategy.java

```diff
@@ -179,6 +180,8 @@ public IncrementalRocksDBSnapshotResources syncPrepareResources(long checkpointI
 
         return new RocksDBIncrementalSnapshotOperation(
                 checkpointId,
+                checkpointOptions.getCheckpointType().getSharingFilesStrategy()
+                        == CheckpointType.SharingFilesStrategy.FORWARD_BACKWARD,
                 checkpointStreamFactory,
                 snapshotResources.snapshotDirectory,
                 snapshotResources.baseSstFiles,
```

Comment:
I've just noticed that this part can be simplified: 
instead of passing and handling `reuseSharedFiles` to `RocksDBIncrementalSnapshotOperation`;
we can just pass empty `baseSstFiles` (if sharingStrategy is not `FORWARD_BACKWARD`).

`RocksDBIncrementalSnapshotOperation` doesn't have to be changed then. WDYT?

Reply:
Sure it works.

My goal was to already provide a solution that could be easier changed to `duplicate` instead of reupload, but as the change is so small it can be done as part of https://issues.apache.org/jira/browse/FLINK-25195.

I changed to passing an empty set for now.